### ISSUE TITLE
✍️ Contribution period: Samted Uche

### ISSUE THREAD

**Issue by teddyCodex** [ersilia-os/ersilia#1017]
### Week 1 - Get to know the community

- [X] Join the communication channels
- [X] Open a GitHub issue (this one!)
- [x] Install the Ersilia Model Hub and test the simplest model
- [x] Install Docker if needed, and test another model
- [x] Write a motivation statement to work at Ersilia
- [x] Submit your first contribution to the Outreachy site

### Week 2 - Get Familiar with Machine Learning for Chemistry

- [x] Select a model from the list suggested in GitBook
- [x] Download and serve the model via the Ersilia Model Hub to ensure it works
- [x] Open a repository on your GitHub user with all the necessary files
- [x] Select and clean a dataset of 1000 molecules (example notebook 1)
- [x] Run predictions for the molecules on the selected model and evaluate the results

### Week 3 - Validate a Model in the Wild

- [ ] Find a suitable dataset with sufficient experimental results
- [ ] Clean and standardize the dataset
- [ ] Run predictions and calculate metrics.

### Week 4 - Prepare your final application

- [ ] Submit the final application in the Outreachy website

**Comment by teddyCodex** [comment]
As a data analyst, I am driven by a profound passion for technology and a desire to contribute to the advancement of artificial intelligence (AI) and machine learning (ML). With a strong foundation in Python programming and PostgreSQL, coupled with professional experience as a data analyst, I possess the relevant skills to make a meaningful impact in this rapidly evolving fields.

The opportunity to join the Outreachy internship program and work with Ersilia is quite exciting. I am captivated by the prospect of working on cutting-edge technologies that have the potential to reshape the way we approach AI and ML.

Through this internship, I aim to immerse myself in Ersilia's innovative ecosystem, expanding my knowledge, refining my skills, and gaining invaluable experience. I envision actively contributing to the codebase, engaging in discussions, and proposing innovative solutions that align with Ersilia's vision. Moreover, I am eager to learn from the wealth of knowledge and experience within the community, fostering a mutually beneficial exchange of ideas and best practices.

Beyond the technical aspects, this internship holds significant importance for my personal and professional growth. It represents a pivotal stepping stone in my career journey, where I can acquire practical knowledge, cultivate problem-solving abilities, and enhance my communication and adaptability skills. By collaborating with experienced professionals and tackling real-world challenges, I will be better equipped to navigate the dynamic landscape of AI and ML.

**Comment by teddyCodex** [comment]
As a data analyst with developing skills in Python, scientific computing, database management, and web development, I am extremely motivated to contribute to the Ersilia Open Source Initiative through the Outreachy internship program. 
I find Ersilia's mission to democratize access to AI/ML models for biomedical research and support neglected disease research in low and middle-income countries quite interesting, and I am inspired by Ersilia's commitment to open science.
With my focus on financial data, I am quite new to the fields of AI and ML. However, as someone who pivoted from accounting to data analysis, I believe that this uncharted territory for me is well within my reach and I believe that working with Ersilia will give me much-needed exposure to the field of AI and ML.

With my solid foundation in data analysis and Python programming, I believe I have the technical skills to make valid contributions to this project. My experience with scientific computing pipelines and databases would allow me to assist in model integration, infrastructure setup, and data management. Additionally, my web development skills could support the creation of user-friendly interfaces for browsing and deploying models.

I am committed to leveraging my skills and dedication to make a positive impact through this internship and I would be honored to play a role in advancing this mission and vision of egalitarian healthcare for all. Thank you for your consideration.

**Comment by teddyCodex** [comment]
Notes
Certain models are only compatible with linux/amd64 processors while some models are compatible with both linux/amd64 and linux/arm64 processors.
Some users may exprience this error when trying to pull non-compatible models:
```
Using default tag: latest
latest: Pulling from ersiliaos/eos2ta5
no matching manifest for linux/arm64/v8 in the manifest list entries
```

**Comment by teddyCodex** [comment]
Selected Model eos30gr for Week 2. I consider its impact very valuable. 
Also, there is support for linux/arm64 processors.
Successfully pulled and tested the model.

**Comment by DhanshreeA** [comment]
> **Notes** Certain models are only compatible with linux/amd64 processors while some models are compatible with both linux/amd64 and linux/arm64 processors. Some users may exprience this error when trying to pull non-compatible models:
> 
> ```
> Using default tag: latest
> latest: Pulling from ersiliaos/eos2ta5
> no matching manifest for linux/arm64/v8 in the manifest list entries
> ```

Good to know, we'll keep a track of this!

**Comment by teddyCodex** [comment]
Using the SMILES dataset provided in the #data channel.
Will experiment later using other datasets in the wild.

**Comment by teddyCodex** [comment]
Encountered error while running `ersilia api run -i input.csv -o output.csv`
Error is as a result of missing isaura installation.

Tried installing isaura using pip but encoutering an error relating to HDF5. 
Error snippet: `Building h5py requires pkg-config unless the HDF5 path is explicitly specified using the environment variable HDF5_DIR. For more information and details, see https://docs.h5py.org/en/stable/build.html#custom-installation
      error: pkg-config probably not installed: FileNotFoundError(2, 'No such file or directory')`

Installing now with brew to see if that resolves the isaura install error.

**Comment by teddyCodex** [comment]
Opted to use the Ersilia python library to serve and run the selected model [eos30gr].

**Note** - Pulls results much faster than the CLI

**Comment by teddyCodex** [comment]
![Uploading Unknown.png…]()

**Comment by teddyCodex** [comment]
![Unknown](https://github.com/ersilia-os/ersilia/assets/15376476/a3f9ec22-696b-4b93-96e1-36dec3f88694)

**Comment by teddyCodex** [comment]
![Unknown](https://github.com/ersilia-os/ersilia/assets/15376476/44580e7f-b4a6-451d-a44a-26b4c31c6940)

**Comment by teddyCodex** [comment]
![Unknown](https://github.com/ersilia-os/ersilia/assets/15376476/aa578f7f-3eed-479f-b03b-9fdfb6e963b5)

**Comment by teddyCodex** [comment]
![Unknown](https://github.com/ersilia-os/ersilia/assets/15376476/8209be52-2edb-454f-a4b7-d1b306d081e2)

The scatter plot provides a visual representation of each molecule's prediction value, with color indicating whether it is a herg blocker or not. The red horizontal line at the 0.8 threshold helps to visually separate predicted here blockers from non-blockers. The scatter plot allows us to see any patterns or clusters in predictions and how well the prediction values correlate with the actual 'is_blocker' labels.

**Comment by DhanshreeA** [comment]
Hi @teddyCodex I only see updates for task 1 from week 2. So far this looks good but I would advise against starting week 3 tasks. Only finish week 2 tasks if you can, I will do a final revision on Monday.

**Comment by teddyCodex** [comment]
Model Reproducibility:

Using the dataset from this [publication](https://pubs.acs.org/doi/full/10.1021/acs.jcim.8b00769), I attempt to reproduce the results achieved in Table S6 which includes detailed predictions for 49 approved antineoplastic drugs (including immunomodulating agents) by deephERG. 
The results are unexpectedly varied

![Unknown](https://github.com/ersilia-os/ersilia/assets/15376476/ed677056-7ab2-42d5-b98c-ce73d2c0b407)
![Unknown2](https://github.com/ersilia-os/ersilia/assets/15376476/08932931-7888-40c6-b32f-332aa76214fd)

The model does poorly in reproducing the results of the original publication.
Thoughts / Recommendations:
- Review and ensure consistency in training and evaluation data of the model.
- Verify that there is no data leakage, where information from the test set is inadvertently included in the training set.

**Comment by teddyCodex** [comment]
I had concerns with the results of the model run with python and wondered if the results would be the same using the CLI
Model produced pretty much the same results as the Python Library.

It appears that the disconnect may come from the training and/or validation data
![Unknown3](https://github.com/ersilia-os/ersilia/assets/15376476/d6f888b6-8952-4a6f-8af6-e65b162b05bf)

**Comment by DhanshreeA** [comment]
Thank you @teddyCodex , this is helpful. We have been observing issues with eos30gr. Your work will help us improve this model. Please go ahead and create your final application on Outreachy. :)
